build/main.go

```diff
@@ -92,6 +99,22 @@ type NativeAmount struct {
 	Amount string
 }
 
+type Path struct {
```

Comment:
IMO, the build `package` should be about ease-of-use.  IMO, building up a big nested object graph using go literals is a little heavy, especially in the case where there is small path involved:

```
Path{ Assets: []Asset{ Asset{Native: true} } }
```

As opposed to one of the following:

```
Path().Native()
Path{}.Native() // I think this is valid provided we use non-pointer method definitions
```

The equivalent of your example send would be:
```
Path().Native().Credit("BTC", "GAHJZHVKFLATAATJH46C7OK2ZOVRD47GZBGQ7P6OCVF6RJDCEG5JMQBQ")
```